src/base/configuration.cpp

```diff
@@ -253,68 +254,38 @@ bool Configuration::isBuiltWithPoly()
   return IS_POLY_BUILD;
 }
 
-unsigned Configuration::getNumDebugTags() {
-#if defined(CVC5_DEBUG) && defined(CVC5_TRACING)
-  /* -1 because a NULL pointer is inserted as the last value */
-  return (sizeof(Debug_tags) / sizeof(Debug_tags[0])) - 1;
-#else  /* CVC5_DEBUG && CVC5_TRACING */
-  return 0;
-#endif /* CVC5_DEBUG && CVC5_TRACING */
-}
-
-char const* const* Configuration::getDebugTags() {
+std::vector<std::string> Configuration::getDebugTags()
```

Comment:
Can we return a `const std::vector<std::string>&` instead and get rid of the `#if` below? Isn't the default an empty vector?

Reply:
Right now, we always generate the full vector in `collect_tags.py` and the whole `#ifdef` stuff is done so that this vector does not end up in the binary if it is not necessary. We could push the `#ifdef` stuff into the generated headers (`Debug_tags.h`)...